app/src/main/java/com/wireguard/android/backend/WgQuickBackend.kt

```diff
@@ -104,23 +101,20 @@ class WgQuickBackend(private var context: Context) : Backend {
     }
 
     override fun postNotification(state: State, tunnel: Tunnel) {
+        val intent = Intent(context, NotificationService::class.java)
+        intent.putExtra("id", tunnel.name.hashCode())
+        intent.putExtra("name", tunnel.name)
         if (state == State.UP) {
-            val intent = Intent(context, MainActivity::class.java)
-            intent.flags = Intent.FLAG_ACTIVITY_CLEAR_TASK or Intent.FLAG_ACTIVITY_NEW_TASK
-            val pendingIntent = PendingIntent.getActivity(context, 0, intent, 0)
-            val builder = NotificationCompat.Builder(
-                    context,
-                    TunnelManager.NOTIFICATION_CHANNEL_ID
-            )
-            builder.setContentTitle(context.getString(R.string.notification_channel_wgquick_title))
-                    .setContentText(tunnel.name)
-                    .setContentIntent(pendingIntent)
-                    .setOngoing(true)
-                    .setPriority(Notification.FLAG_ONGOING_EVENT)
-                    .setSmallIcon(R.drawable.ic_qs_tile)
-            notificationManager.notify(tunnel.name.hashCode(), builder.build())
+            intent.putExtra("isServiceRunning", false)
```

Comment:
Do you need to send this? You can easily retrieve state by checking active tunnels